syntax = "proto3";

package user_service;

service UserService {
    rpc CreateUser (User) returns (UserResponse);
    rpc GetUser (UserRequest) returns (UserResponse);
    rpc ListUsers (Empty) returns (UserList);
}

message User {
    int32 id = 1;
    string username = 2;
    string email = 3;
    string password = 4;
}

message UserRequest {
    int32 id = 1;
}

message UserResponse {
    User user = 1;
    string error = 2;
}

message UserList {
    repeated User users = 1;
}

message Empty {}